File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.stream.*;
import java.util.*;
import java.util.Map.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> map = new HashMap();
        for (int i : tangerine){
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        
        List<Entry<Integer,Integer>> entry = map.entrySet()
            .stream().sorted((a,b)->b.getValue() - a.getValue())
            .collect(Collectors.toList());
        
        int cnt = 0;
        for(Entry<Integer,Integer> e : entry){
            cnt += 1;
            k -= e.getValue();
            if (k <= 0) break;
        }
        return cnt;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let map = {}
    for(let t of tangerine){
        map[t] = (map[t] ?? 0) + 1;
    }
    
    let values = Object.values(map).sort((a,b) => a - b);
    
    let cnt = 0;
    while (k > 0){
        k -= values.pop();
        cnt += 1
    }
    
    return cnt;
}